src/App.js

```diff
@@ -1,16 +1,45 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog'
+
 
 const App = () => {
+  const [chatData, setChatmessages] = useState(chatMessages)
+
+  const updateChatData = updatedChat => {
+    const chats = chatData.map(chat => {
+      if (chat.id === updatedChat.id) {
+        return updatedChat;
```

Comment:
If you refactor this method so that it takes in an id as a parameter, then you'd need to update line 13 to make a copy of the object with the matching id and switch the value of the liked here in app.js (instead of in ChatEntry)